tests/core_tests/validations/base_functions.py

```diff
@@ -0,0 +1,111 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2020 - 2023 Pionix GmbH and Contributors to EVerest
+
+import logging
+import time
+import asyncio
+from typing import Callable
+from typing import Any
+
+from everest.testing.core_utils.test_control_module import TestControlModule
+
+## helper functions
+
+## ---------------------------------------------------
+def get_key_if_exists(dataset:dict, keyset:list):
```

Comment:
syntax should be `dataset: dict, keyset: list`
the naming is a bit misleading,  dataset could be called just dictionary
keyset is really only a key.
Actually this function could be just replaced by `dict.get(key, None)`, of course if the value is allowed to be None, then one would need to check for `key in dict`